### ISSUE TITLE
Padding Masks Option

### ISSUE THREAD

**Issue by yelnady** [zhouhaoyi/Informer2020#104]
Hi,

Thanks for sharing the code of this powerful paper, and the results are very interesting. 
I have a question regarding the **padding masks.** In the PyTorch (**torch.nn.TransformerEncoder**), there's an option for **src_key_padding_mask** and **mask**. I wonder if your model supports padding masks or not, which is different from the square matrix mask. Basically, I want to use your model for variable-length sequential data where the samples have different lengths. 

torch.nn.TransformerEncoder: https://pytorch.org/docs/stable/generated/torch.nn.TransformerEncoder.html#torch.nn.TransformerEncoder

**Comment by cookieminions** [comment]
Hi,

Thanks for your attention to our work.
Our model support mask, and you can use mask by settting mask_flag to True in code `models/model.py`

https://github.com/zhouhaoyi/Informer2020/blob/410535ebeb59e526a7402e0454235fe1cab83605/models/attn.py#L38-L40

https://github.com/zhouhaoyi/Informer2020/blob/410535ebeb59e526a7402e0454235fe1cab83605/models/model.py#L31-L32

The target of Informer is time series forecasting, and the input length is fixed in our experiments. But parameters of our model have nothing to do with the length of the input sequence, so it's okay to feed sequences with different lengths to our model.

**Comment by yelnady** [comment]
Thank you so much!
